docs/DeveloperGuide.md

```diff
@@ -1,38 +1,361 @@
+
+Given below is an example of the usage of this feature and the mechanism at each step
+
+Step 1: The user launches the application for the first time and `AccountList` is created with no `Account`
+![AccountListObjectDiagram1](images/AccountListObjectDiagram1.png)
+
+Step 2: The user passes in the command `create-account <CURRENCY>`, where `CURRENCY` is a valid string representing one
+of the elements of the `Currency` enum
+![AccountListObjectDiagram1](images/AccountListObjectDiagram2.png)
+
+Step 3: The user passes in the command `create-account <CURRENCY>`, where `CURRENCY` is also valid but different to that
+in step 1.
+![AccountListObjectDiagram1](images/AccountListObjectDiagram3.png)
+
+Step 4: The user passes the command `delete-account CURRENCY`, for example `delete-account SGD`.
+
+![AccountListObjectDiagram4](images/AccountListObjectDiagram4.png)
+
+The following sequence diagram shows how the Create Account operation works
+![AccountListObjectDiagram1](images/CreateAccountSeqDiagram.png)
+
+### Delete-account feature
+
+### Add/Withdraw money feature
+
+The add money(deposit) and withdraw money feature is facilitated by `AddCommand` and `WithdrawCommand` which both
+extends the `Command` class. With the provided input from user (`CURRENCY` and `AMOUNT`), `AddCommand`
+and `WithdrawCommand`
+update the balance of respective currency account accordingly.
+
+Step 1. The newly created `SGD` account has an initial balance of 0
+![AddWithdrawCommandObjectDiagram1](images/AddWithdrawCommandObjectDiagram1.png)
+
+Step 2. The user passes command `add CURRENCY AMOUNT` (eg. `add SGD 100`), where `CURRENCY` must be one of the available
+currency and `AMOUNT` must be positive numbers.
+![AddWithdrawCommandObjectDiagram2](images/AddWithdrawCommandObjectDiagram2.png)
+
+Step 3. The user passes command `withdraw` (eg. `withdraw SGD 25`), where `AMOUNT` must be smaller than the currency
+account balance.
+![AddWithdrawCommandObjectDiagram3](images/AddWithdrawCommandObjectDiagram3.png)
+
+The following sequence diagram shows how the add money operation works.
+
+![](images/AddCommandSeqDiagram.png)
+
+The following sequence diagram shows how the money withdrawal operation works.
+
+![](images/WithdrawCommandSeqDiagram.png)
+
+### View balance feature
+
+The view balance feature is facilitated using `Account` instances stored within the `AccountList` object. The main
+functionality is to view the balance of a specific currency if the currency is specified, else view all the currencies
+in the account.
+
+![BalanceSequenceDiagram](images/BalanceSeqDiagram.png)
+
+### Show-rate feature
+
+The exchange feature is facilitated using two `Forex` instances to represent the
+exchange rates between two currencies. The current implementation reads manual exchange
+rates from an online source. Future implementation will use an API to maintain up-to-date
+exchange rates.
+
+Exchange rate source: https://www.xe.com/currencyconverter/convert
```

Comment:
Could you consider making this a link instead of pasting it as text?
<img width="719" alt="image" src="https://user-images.githubusercontent.com/91600220/229009881-607ef718-48ce-473b-b304-7213e0747012.png">